yarn/src/main/scala/org/apache/spark/deploy/yarn/Client.scala

```diff
@@ -55,13 +57,9 @@ private[spark] class Client(
    * ------------------------------------------------------------------------------------- */
 
   /**
-   * Submit an application running our ApplicationMaster to the ResourceManager.
-   *
-   * The stable Yarn API provides a convenience method (YarnClient#createApplication) for
-   * creating applications and setting up the application submission context. This was not
-   * available in the alpha API.
+   * Create an application running our ApplicationMaster to the ResourceManager.
    */
-  override def submitApplication(): ApplicationId = {
+  override def createApplication(): ApplicationId = {
```

Comment:
SparkContext firstly gets applicationId from taskScheduler and uses it to initialize blockManager and eventLogger. And then dagScheduler runs job and submits resources requests to cluster master.
Getting applicationId and submitting resources requests to cluster master are split into two methods.